modules/javafx.graphics/src/main/java/com/sun/glass/utils/NativeLibLoader.java

```diff
@@ -222,15 +222,34 @@ private static boolean installLibraryFromResource(String libraryName, List<Strin
 
     private static String cacheLibrary(InputStream is, String name, Class caller) throws IOException {
         String jfxVersion = System.getProperty("javafx.version", "versionless");
-        String userCache = System.getProperty("user.home") + "/.openjfx/cache/" + jfxVersion;
+        String userCache = System.getProperty("javafx.cachedir", "");
+        if (userCache.isEmpty()) {
+            userCache = System.getProperty("user.home") + "/.openjfx/cache/" + jfxVersion;
+        }
         File cacheDir = new File(userCache);
+        boolean cacheDirOk = true;
         if (cacheDir.exists()) {
             if (!cacheDir.isDirectory()) {
-                throw new IOException ("Cache exists but is not a directory: "+cacheDir);
+                System.err.println("Cache exists but is not a directory: "+cacheDir);
+                cacheDirOk = false;
             }
         } else {
             if (!cacheDir.mkdirs()) {
-                throw new IOException ("Can not create cache at "+cacheDir);
+                System.err.println("Can not create cache at "+cacheDir);
+                cacheDirOk = false;
+            }
+        }
+        if (!cacheDirOk) {
+            String tmpCache = System.getProperty("java.io.tmpdir") + "/.openjfx/cache/" + jfxVersion;
```

Comment:
Do you think appending the username might be useful? Otherwise it may not work for multiple users accessing the same directory.

Reply:
I added the username.

Reply:
May I suggest auto generating a sub directory within the java.io.tmpdir instead?

Reply:
Using createTempFile?

Reply:
I think although in the case the user.home is for some reason not applicable and we use the tmp directory, the directory we create should not really be considered a temporarily directory. 
When the process exists, there is no need to remove the directory, at the contrary (so we don't need the hooks for deleting on exit etc)